### PR TITLE
Upgrade firecracker to 1.4.1

### PR DESCRIPTION
**Pull request by fangn2**
*Issue #, if available:*
Upgrade firecracker from 1.0.0 to 1.4.1

*Description of changes:*
Upgrade firecracker to 1.4.1 
https://github.com/firecracker-microvm/firecracker/compare/v1.0.0...v1.4.1
PR #495 has upgraded firecracker partially to 1.4.1(only snapshot memory backend was added, not all swagger specs were in sync with v1.4.1), this PR is a followup to make sure swagger files are in sync and client were generated. 

1. Pull swagger specs from firecracker 1.4.1
2. `go generate` to rebuild the client
3. Upgrade version in the Makefile

Missing features not yet implemented at sdk side are captured in the [list](https://github.com/firecracker-microvm/firecracker-go-sdk/issues?q=is%3Aissue+is%3Aopen+label%3Afirecracker-feature).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
Sign-off-by: Tony Fang <nhfang@amazon.com>

### CODE REVIEW

File: examples/cmd/snapshotting/Makefile
Comment (ginglis13) [inline examples/cmd/snapshotting/Makefile:64]: do we need this hardcoded? won't the same issue occur the next time firecracker upgrades the Ubuntu version for building rootfs w/ devtool? not pretty, but could we just cp `*.rootfs.ext4` since the devtool only makes 1 rootfs?

Comment (fangn2) [reply examples/cmd/snapshotting/Makefile:64]: Yeah, make sense. 
I do have a comment message above mentioning we need to change this if firecracker uses different Ubuntu distro.
I can change it the `*.rootfs.ext4`, previously I am not sure if there are other rootfs.

**Comment by fangn2** [comment]
Waiting for #495 to merge to unblock build issue.

**Comment by bduffany** [comment]
> Waiting for #495 to merge to unblock build issue.

I think my PR (#495) is dependent on upgrading Firecracker, since `MemoryBackend` is not available in Firecracker v1.0.0.

**Comment by fangn2** [comment]
>I can bump firecracker_version in the Makefile in my PR, if that's all that's needed. Let me know if I need to do anything else.

Thanks! From test perspetive, change that is enough to test with firecracker 1.4.0

File: client/swagger.yaml
Comment (swagatbora90) [inline client/swagger.yaml:198]: this is just updating the fc client apis, we still need to add support for this feature in the go-sdk. Can we add a issue to track the compatibility work?

Comment (fangn2) [reply client/swagger.yaml:198]: Created two issues #505 and #504 to track unimplemented features introduced from the upgrade.

File: Makefile
Comment (swagatbora90) [inline Makefile:32]: Do you want to move directly to 1.4.1 https://github.com/firecracker-microvm/firecracker/releases/tag/v1.4.1?

Comment (fangn2) [reply Makefile:32]: Sure.

**Comment by fangn2** [comment]
Since firecracker 1.4.1 was released, upgrade to it directly. 
1.4.0 to 1.4.1 has no swagger.yaml change.

**Review by swagatbora90 — APPROVED** [review]
LGTM

**Review by ginglis13 — APPROVED** [review]

**Pull request merged**
